### PR TITLE
feature/#163 added doc for running core rest and gui together

### PR DESCRIPTION
**Pull request by toan-quach**
#163

### CODE REVIEW

**Comment by jrobinAV** [comment]
As discussed, you should: 
- move the text you wrote on deployment to the new section "Running Taipy services" at the top level.
- create a small note or a reminder block to say something like `for running X service with the other services, please refer to page Running Taipy services` every time a run method is documented.

File: docs/manuals/gui/configuration.md [hunk 1]
```diff
@@ -110,6 +110,8 @@ Here is the list of the configuration parameters you can use in
      application to the Internet. See the section on
      [Accessing your app from the Web](#accessing-your-app-from-the-web) for details.
 
+!!! info "For running Taipy Gui service with the other Taipy services, please refer to page [Running Taipy services](../running_services/index.md)"
```
Comment (jrobinAV) [inline docs/manuals/gui/configuration.md]:
```suggestion
!!! info "For running Taipy Gui service with some other Taipy services, please refer to page [Running Taipy services](../running_services/index.md)"
```

File: docs/manuals/gui/index.md [hunk 2]
```diff
@@ -45,3 +45,4 @@ content sent back to the client so the user can see the application interface
 and start using it.
 
 !!! info "You can find more information on how pages are created and used in Taipy application in the [Pages](pages.md) section."
+!!! info "For running Taipy Gui service with the other Taipy services, please refer to page [Running Taipy services](../running_services/index.md)"
```
Comment (jrobinAV) [inline docs/manuals/gui/index.md]:
```suggestion
!!! info "For running Taipy Gui service with some other Taipy services, please refer to page [Running Taipy services](../running_services/index.md)"
```

File: docs/manuals/running_services/index.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+# Approach 1
+core.run()
+
+# Approach 2
+tp.run([core])
+```
+
+# Runing different Taipy services together
+
+You can run different Taipy services together by using `taipy.run()^`.
```
Comment (jrobinAV) [inline docs/manuals/running_services/index.md]:
```suggestion
You can run various Taipy services together by using `taipy.run()^`.
```

File: docs/manuals/running_services/index.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+Since running `Rest^` means you will also be running `Core^`, the following code is the same as the above:
+```python
+import taipy as tp
+
+gui = tp.Gui(page="# Getting started with *Taipy*")
+rest = tp.Rest()
+
+tp.run([gui, rest], title="Taipy Demo")
+```
```
Comment (jrobinAV) [inline docs/manuals/running_services/index.md]: I believe we only need to have 2 examples : 
- GUI + Core
- GUI + Rest

**Review by jrobinAV — COMMENTED** [review]
We have another ticket related to running taipy services: [taipy-59](https://app.zenhub.com/workspaces/taipy-613ef328affde6000ff77fb3/issues/avaiga/taipy/59.)

To summarize, we need to document that the application initialization code and run method should be called in the main block.

Can you change your code examples to add the main block definition? something like:
 
```
if __name__ == "__main__":
    tp.run(gui, rest)
```

File: docs/manuals/running_services/index.md (hunk 4 above)
Comment (toan-quach) [reply docs/manuals/running_services/index.md]: I have removed the part where we call tp.run([core, rest, gui]) as I think it's duplicated. However, I still think it's necessary to mention that we will be running Core everytime we run Rest, and so I'll move it up to running individual services. What do you think?

File: docs/manuals/gui/configuration.md [hunk 5]
```diff
@@ -110,6 +110,8 @@ Here is the list of the configuration parameters you can use in
      application to the Internet. See the section on
      [Accessing your app from the Web](#accessing-your-app-from-the-web) for details.
 
+!!! info "For running Taipy Gui service with some other Taipy services, please refer to page [Running Taipy services](../running_services/index.md)"
```
Comment (FabienLelaquais) [inline docs/manuals/gui/configuration.md]: "For running Taipy Gui service"
->
"To run the Taipy GUI service"

"to page [Running Taipy services](../running_services/index.md)"
to
"to the [Running Taipy services](../running_services/index.md) page"

File: docs/manuals/gui/index.md [hunk 6]
```diff
@@ -45,3 +45,4 @@ content sent back to the client so the user can see the application interface
 and start using it.
 
 !!! info "You can find more information on how pages are created and used in Taipy application in the [Pages](pages.md) section."
+!!! info "For running Taipy Gui service with some other Taipy services, please refer to page [Running Taipy services](../running_services/index.md)"
```
Comment (FabienLelaquais) [inline docs/manuals/gui/index.md]: "For running Taipy Gui service"
->
"To run the Taipy GUI service"

"to page [Running Taipy services](https://github.com/Avaiga/taipy-doc/pull/running_services/index.md)"
to
"to the [Running Taipy services](https://github.com/Avaiga/taipy-doc/pull/running_services/index.md) page"

File: docs/manuals/gui/pages.md [hunk 7]
```diff
@@ -232,6 +232,8 @@ for all its pages.
     If you navigate to '/page2', the main title remains on the page, and the sub-title
     is replaced by the text 'This is page 2'
 
+    !!! info "For running Taipy Gui service with the other Taipy services, please refer to page [Running Taipy services](../running_services/index.md)"
```
Comment (FabienLelaquais) [inline docs/manuals/gui/pages.md]: "For running Taipy Gui service"
->
"To run the Taipy GUI service"

"to page [Running Taipy services](https://github.com/Avaiga/taipy-doc/pull/running_services/index.md)"
to
"to the [Running Taipy services](https://github.com/Avaiga/taipy-doc/pull/running_services/index.md) page"

File: docs/manuals/rest/index.md [hunk 8]
```diff
@@ -48,6 +48,9 @@ To expose the Taipy REST APIs, the Taipy REST server must first be started.
    * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
    ```
 
+!!! Note "When running Taipy Rest server, you will also run `Core^`"
```
Comment (FabienLelaquais) [inline docs/manuals/rest/index.md]: "running Taipy Rest server"
->
"running the Taipy Rest server"

File: docs/manuals/rest/index.md [hunk 9]
```diff
@@ -48,6 +48,9 @@ To expose the Taipy REST APIs, the Taipy REST server must first be started.
...
+!!! info "For running Taipy Rest service with the other Taipy services, please refer to page [Running Taipy services](../running_services/index.md)"
```
Comment (FabienLelaquais) [inline docs/manuals/rest/index.md]: "For running Taipy Gui service"
->
"To run the Taipy GUI service"

"to page [Running Taipy services](https://github.com/Avaiga/taipy-doc/pull/running_services/index.md)"
to
"to the [Running Taipy services](https://github.com/Avaiga/taipy-doc/pull/running_services/index.md) page"

File: docs/manuals/running_services/index.md [hunk 10]
```diff
@@ -0,0 +1,90 @@
+# Running Taipy services
+
+Taipy provides you three services: Taipy Gui, Taipy Rest, and Taipy Core. You can run Taipy services by calling the method _run()_ from the service instance either `Gui^`, `Rest^`, or `Core^`. You can also use `taipy.run()^` to initiate the service.
```
Comment (FabienLelaquais) [inline docs/manuals/running_services/index.md]: Gui -> GUI

"from the service instance either `Gui^`, `Rest^`, or `Core^`"
->
"from of a service instance that can be either `Gui^`, `Rest^`, or `Core^`"

"to initiate the service"
->
"to initiate services"

File: docs/manuals/running_services/index.md [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+
+For example, you can run the Gui service with the following code:
```
Comment (FabienLelaquais) [inline docs/manuals/running_services/index.md]: Gui -> GUI

File: docs/manuals/running_services/index.md [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+    gui = tp.Gui(page="# Getting started with *Taipy*")
+
+    # Approach 1
+    gui.run(gui, title="Taipy Demo")
+
+    # Approach 2
+    tp.run([gui], title="Taipy Demo")
+```
+
+By default, when running the Rest service, you will also start the Core service as the Rest service relies on the Core service to function:
```
Comment (FabienLelaquais) [inline docs/manuals/running_services/index.md:19]: Remove "to function"

File: docs/manuals/running_services/index.md [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+import taipy as tp
+
+if __name__ == "__main__":
+    gui = tp.Gui(page="# Getting started with *Taipy*")
+
+    # Approach 1
+    gui.run(gui, title="Taipy Demo")
+
+    # Approach 2
+    tp.run([gui], title="Taipy Demo")
```
Comment (FabienLelaquais) [inline docs/manuals/running_services/index.md]: Is it possible to remove the mandatory array creation?

File: docs/manuals/running_services/index.md [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+import taipy as tp
+
+if __name__ == "__main__":
+    rest = tp.Rest()
+
+    # Approach 1
+    rest.run()
+
+    # Approach 2
+    tp.run([rest])
```
Comment (FabienLelaquais) [inline docs/manuals/running_services/index.md]: Is it possible to remove the mandatory array creation?

File: docs/manuals/running_services/index.md [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+
+    # Approach 1
+    core.run()
+    rest.run()
+
+    # Approach 2
+    tp.run([core, rest], title="Taipy Demo")
+```
+
+You can run the Core service individually with the below code:
```
Comment (FabienLelaquais) [inline docs/manuals/running_services/index.md]: "below code" -> "code below"

File: docs/manuals/running_services/index.md [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+import taipy as tp
+
+if __name__ == "__main__":
+    core = tp.Core()
+
+    # Approach 1
+    core.run()
+
+    # Approach 2
+    tp.run([core])
```
Comment (FabienLelaquais) [inline docs/manuals/running_services/index.md]: Is it possible to remove the mandatory array creation?

File: docs/manuals/running_services/index.md [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+```
+
+# Runing different Taipy services together
```
Comment (FabienLelaquais) [inline docs/manuals/running_services/index.md]: "Runing" -> "Running"

File: docs/manuals/running_services/index.md [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+
+You can run various Taipy services together by using `taipy.run()^`.
+
+You can run Taipy Gui along with Taipy Core together with the following code:
```
Comment (FabienLelaquais) [inline docs/manuals/running_services/index.md]: Gui -> GUI

File: docs/manuals/running_services/index.md [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+import taipy as tp
+
+if __name__ == "__main__":
+    gui = tp.Gui(page="# Getting started with *Taipy*")
+    core = tp.Core()
+
+    tp.run([gui, core], title="Taipy Demo")
+```
+
+You can run all Taipy services including Taipy Gui, Taipy Rest, and Taipy Core with the below code:
```
Comment (FabienLelaquais) [inline docs/manuals/running_services/index.md]: Gui -> GUI

"below code" -> "code below"

File: docs/manuals/running_services/index.md [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+import taipy as tp
+
+if __name__ == "__main__":
+    gui = tp.Gui(page="# Getting started with *Taipy*")
+    rest = tp.Rest()
+
+    tp.run([gui, rest], title="Taipy Demo")
+```
+
+!!! note "As mentioned, Taipy Rest needs Taipy Core to function, when running the Taipy Rest service, Taipy Core will also run along with it. Therefore, Taipy Core will also be together with Taipy Gui and Taipy Rest in the above example."
```
Comment (FabienLelaquais) [inline docs/manuals/running_services/index.md]: "Taipy Rest needs Taipy Core to function, when running"
->
"Taipy Rest relies on Taipy Core: when running" 

"Therefore, Taipy Core will also be together with Taipy Gui and Taipy Rest in the above example."
I don't understand this sentence.

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
A few typos and a question...

File: docs/manuals/running_services/index.md (hunk 20 above)
Comment (toan-quach) [reply docs/manuals/running_services/index.md]: @FabienLelaquais oh it should be "Therefore, Taipy Core will also be run together with Taipy Gui and Taipy Rest in the above example."

File: docs/manuals/running_services/index.md (hunk 16 above)
Comment (toan-quach) [reply docs/manuals/running_services/index.md]: @FabienLelaquais ah yes you can! I'll remove the array creation then

File: docs/manuals/running_services/index.md (hunk 20 above)
Comment (FabienLelaquais) [reply docs/manuals/running_services/index.md]: Except for 'Gui' being spelled 'GUI', I'm good with that

File: docs/manuals/running_services/index.md (hunk 4 above)
Comment (jrobinAV) [reply docs/manuals/running_services/index.md]: Seems good!

File: docs/manuals/rest/index.md [hunk 21]
```diff
@@ -48,8 +48,8 @@ To expose the Taipy REST APIs, the Taipy REST server must first be started.
    * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
    ```
 
-!!! Note "When running Taipy Rest server, you will also run `Core^`"
-!!! info "For running Taipy Rest service with the other Taipy services, please refer to page [Running Taipy services](../running_services/index.md)"
+!!! Note "When running the Taipy Rest server, you will also run `Core^`"
+!!! info "To run the Taipy Rest service with the other Taipy services, please refer to the [Running Taipy services](../running_services/index.md)" page
```
Comment (FabienLelaquais) [inline docs/manuals/rest/index.md]: Misplaced double quote.

File: docs/manuals/running_services/index.md [hunk 22]
```diff
@@ -13,7 +13,7 @@ if __name__ == "__main__":
     gui.run(gui, title="Taipy Demo")
 
     # Approach 2
-    tp.run([gui], title="Taipy Demo")
+    tp.run(gui, title="Taipy Demo")
```
Comment (FabienLelaquais) [inline docs/manuals/running_services/index.md:16]: Thanks!

File: docs/manuals/running_services/index.md [hunk 23]
```diff
@@ -43,10 +43,10 @@ if __name__ == "__main__":
     rest.run()
 
     # Approach 2
-    tp.run([core, rest], title="Taipy Demo")
+    tp.run(core, rest, title="Taipy Demo")
```
Comment (FabienLelaquais) [inline docs/manuals/running_services/index.md:46]: I haven't seen this syntax yet. But it it works, it works.
In the situation where you have more than one service, the array syntax is not such a bit deal to me.

**Review by FabienLelaquais — APPROVED** [review]
Caution: tiny but significant typo.

File: docs/manuals/running_services/index.md (hunk 23 above)
Comment (toan-quach) [reply docs/manuals/running_services/index.md:46]: I have tested and checked the code again, it works for both with array or no array so we should be good :D

File: docs/manuals/rest/index.md (hunk 21 above)
Comment (toan-quach) [reply docs/manuals/rest/index.md]: Thanks!

File: docs/manuals/running_services/index.md (hunk 20 above)
Comment (toan-quach) [reply docs/manuals/running_services/index.md]: Have fixed all of that too! Thanks!

**Pull request merged**